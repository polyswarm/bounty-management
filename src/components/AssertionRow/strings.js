const strings = {
  good: 'Benign',
  bad: 'Malignant',
};

export default strings;
