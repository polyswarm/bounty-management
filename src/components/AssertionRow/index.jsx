import React, { Component } from 'react';
import PropTypes from 'prop-types';
// Component Imports
import strings from './strings';
import './styles.css';

class AssertionRow extends Component {

  render() {
    const { props: { assertion } } = this;
    const verdictClass = `Assertion-${AssertionRow.computeClass(assertion.verdict)}`;
    const verdictString = assertion.verdict ? strings.bad : strings.good;
    return (
      <tr className='AssertionRow'>
        <td>{assertion.author}</td>
        <td className={verdictClass}>{verdictString}</td>
        <td>{assertion.metadata}</td>
        <td>{assertion.bid}</td>
      </tr>
    );
  }

  static computeClass(verdict) {
    if (verdict) {
      return 'Malignant';
    } else {
      return 'Benign';
    }
  }

}

AssertionRow.proptypes = {
  assertion: PropTypes.object.isRequired,
};
export default AssertionRow;
