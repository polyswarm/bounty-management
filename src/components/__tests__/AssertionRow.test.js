import React from 'react';
import {render} from 'enzyme';
import {renderToJson} from 'enzyme-to-json';
import AssertionRow from '../AssertionRow';

it('renders without crashing', () => {
  const assertion = {
    name: 'asdf',
    verdict: true,
    bid: 10,
    metadata: 'Some virus'
  };
  const wrapper = render(
    <table>
      <AssertionRow assertion={assertion}/>
    </table>
  );
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

it('returns Malignant from computeClass with true verdict', () => {
  const result = AssertionRow.computeClass(true);

  expect(result).toEqual('Malignant');
});

it('returns Benign from computeClass with false verdict', () => {
  const result = AssertionRow.computeClass(false);

  expect(result).toEqual('Benign');
});

it('shows Malignant in the veridict column with true verdict', () => {
  const assertion = {
    name: 'asdf',
    verdict: true,
    bid: 10,
    metadata: 'Some virus'
  };
  const wrapper = render(
    <table>
      <AssertionRow assertion={assertion}/>
    </table>
  );

  expect(wrapper.find('.Assertion-Malignant').text()).toEqual('Malicious');
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

it('shows Benign in the veridict column with false verdict', () => {
  const assertion = {
    name: 'asdf',
    verdict: false,
    bid: 10,
    metadata: ''
  };
  const wrapper = render(
    <table>
      <AssertionRow assertion={assertion}/>
    </table>
  );

  expect(wrapper.find('.Assertion-Benign').text()).toEqual('Safe');
  expect(renderToJson(wrapper)).toMatchSnapshot();
});
