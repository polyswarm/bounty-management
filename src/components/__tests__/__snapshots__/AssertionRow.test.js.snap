// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<table>
  <tbody>
    <tr>
      <td>
        
      </td>
      <td
        class="Assertion-Malignant"
      >
        Malignant
      </td>
      <td>
        Some virus
      </td>
      <td>
        10
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`shows Benign in the veridict column with false verdict 1`] = `
<table>
  <tbody>
    <tr>
      <td>
        
      </td>
      <td
        class="Assertion-Benign"
      >
        Benign
      </td>
      <td>
        
      </td>
      <td>
        10
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`shows Malignant in the veridict column with true verdict 1`] = `
<table>
  <tbody>
    <tr>
      <td>
        
      </td>
      <td
        class="Assertion-Malignant"
      >
        Malignant
      </td>
      <td>
        Some virus
      </td>
      <td>
        10
      </td>
    </tr>
  </tbody>
</table>
`;
